### ISSUE TITLE
Bug: `TypeError: Plugin "n" not found.` with 18.2.1 release

### ISSUE THREAD

**Issue by mriedem** [eslint-community/eslint-plugin-n#551]
### Environment

Node version: `v24.11.1`
npm version: `11.6.2`
ESLint version: `10.6.0`
eslint-plugin-n version: `18.2.1`
Operating System: ubuntu linux

### What rule do you want to report?

n/a

### Link to Minimal Reproducible Example

n/a

### What did you expect to happen?

eslint config setup should continue to work; this is our `eslint.config.js` file:

```js
const { defineConfig, globalIgnores } = require('eslint/config')
const js = require('@eslint/js')
const jest = require('eslint-plugin-jest')
const importPlugin = require('eslint-plugin-import-x')
const { default: nodePlugin } = require('eslint-plugin-n')
const eslintPluginPrettierRecommended = require('eslint-plugin-prettier/recommended')
const globals = require('globals')

module.exports = defineConfig([
  globalIgnores([
    '**/docs',
    // This is locally built docs
    '**/site',
    // The db-migrate generated js template files don't conform to our lint rules.
    'migrations/*.js',
    // These k6 performance tests are no longer maintained and should be removed...
    'tests/performance/*.js'
  ]),
  js.configs.recommended,
  importPlugin.flatConfigs.recommended,
  {
    plugins: { n: nodePlugin },
    extends: ['n/recommended-module']
  },
  {
    files: ['tests/**'],
    ...jest.configs['flat/recommended']
  },
  {
    files: ['**/*.test.js'],
    ...jest.configs['flat/style']
  },
  {
    plugins: { jest }, // needed to override jest/expect-expect
    rules: {
      'no-process-exit': 'off',
      'max-len': ['error', 120],
      'jest/expect-expect': ['error', { assertFunctionNames: ['expect*'] }]
    }
  },
  {
    files: ['eslint.config.js', '**/*.test.js', 'tests/routers/helpers.js', 'tests/setWaitForTimeout.js'],
    rules: {
      // This is needed because we use @quantum/wait-for, jsonwebtoken, supertest, and nock.
      // https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/no-unpublished-import.md
      'n/no-unpublished-require': 'off'
    }
  },
  {
    files: ['scripts/*.js'],
    rules: {
      'n/no-process-exit': 'off',
      'n/no-unpublished-require': 'off'
    }
  },
  {
    languageOptions: {
      globals: {
        ...globals.node
      }
    }
  },
  // Any other config imports go at the top so that eslint-config-prettier has
  // the opportunity to override other configs
  eslintPluginPrettierRecommended
])
```

We get this error with 18.2.1 but 18.1.0 works:

```
> multi-channel-scheduler@1.132.0 lint
> eslint .
Oops! Something went wrong! :(
ESLint: 10.6.0
TypeError: Plugin "n" not found.
    at findPluginConfig (/workspace/app/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:263:9)
    at /workspace/app/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:432:25
    at [Array.map](http://array.map/) (<anonymous>)
    at processExtends (/workspace/app/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:430:36)
    at /workspace/app/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:506:38
    at Array.flatMap (<anonymous>)
    at processConfigList (/workspace/app/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:506:20)
    at defineConfig (/workspace/app/multi-channel-scheduler/node_modules/@eslint/config-helpers/dist/cjs/index.cjs:542:9)
    at Object.<anonymous> (/workspace/app/multi-channel-scheduler/eslint.config.js:9:18)
    at Module._compile (node:internal/modules/cjs/loader:1760:14)
=== Execution of custom script for stage 'unit-test' finished. Exit Code: '2'.
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mriedem** [comment]
I also tried this from the FAQ but get the same error:

```js
  {
    plugins: { n: nodePlugin },
    extends: ["n/mixed-esm-and-cjs"]
  },
```

**Comment by aladdin-add** [comment]
v18 is esm-only. if you are still using cjs, please add `.default`:

```js
const nodePlugin = require("eslint-plugin-n").default;
```

**Comment by aladdin-add** [comment]
Following the introduction of the change: https://github.com/eslint-community/eslint-plugin-n/pull/542, `.default` is no longer needed.

```js
const nodePlugin = require('eslint-plugin-n')
```

**Comment by mriedem** [comment]
> Following the introduction of the change: [#542](https://github.com/eslint-community/eslint-plugin-n/pull/542), `.default` is no longer needed.
> 
> const nodePlugin = require('eslint-plugin-n')

Thanks that was our issue.
